lessons/Data-Science/CODAP/codap-composing/langs/en-us/index.adoc

```diff
@@ -0,0 +1,195 @@
+
+[cols= "10,10,10",options="header"]
+|===
+| Filter
+| Build Attribute
+| Transform Attribute
+
+| 1 Transformer Type
+
+2 Dataset
+
+3 Expression
+
+| 1 Transformer Type
+
+2 Dataset
+
+3 Name of New Attribute
+
+4 Expression
+
+| 1 Transformer Type
+
+2 Dataset
+
+3 Attribute to Transform
+
+4 New Name for Transformed Attribute
+
+5 Expression
+
+|===
+
+- How many arguments does Filter require? How about Build Attribute? Transform Attribute?
+** _Filter requires 2 arguments. Build Attribute requires 3 arguments. Transform Attribute requires 4 arguments._
+- Why does each Transformer require a different number of arguments?
+** _All Transformeres require a Transformer type and a dataset. Beyond that, Filter only needs a Boolean to know which rows to include and exclude from the new dataset. Build and Transform both require a new attribute name. Transform also needs to know which attribute is being transformed._
+--
+
+=== Investigate
+
+In the subsequent activities, "Transformer Cards" are a tool for students to think about what specific arguments each Transformer requires. These cards will later support students in composing Transformers.
+
+Let's practice creating Transformer Cards to represent various Transformers.
+
+[.lesson-instruction]
+- Turn to @printable-exercise{create-transformer-cards.adoc}.
+- Respond to the prompts, drawing Circles of Evaluation that represent the transformation needed to create the desired table.
+- In the last column, give each Transformer a useful and descriptive name.
+- Now, complete the handout @handout{create-transformer-cards-2.adoc}. On this page, you will need to determine how many arguments each Transformer requires before creating the appropriate card.
+
+Confirm that students have correct responses on their Transformer Cards, as these cards are central to the subsequent segment of the lesson.
+
+
+=== Synthesize
+
+- Did you get more comfortable creating Transformer cards as you completed these two pages?
+- What obstacles did you encounter?
+- What is the value of naming Transformers?
+
+
+== Composing Transformers @duration{25 minutes}
+
+=== Overview
+
+Students learn to diagram composed Transformers using the Circles of Evaluation. This tool has deep roots in both @lesson-link{order-of-operations} and @lesson-link{function-composition}, and math teachers may want to take a detour through one or both of these lessons to support those learning goals.
+
+
+=== Launch
+
+We already know how to filter, transform, and build columns - but what if we want to do _multiple things, all at once?_ Filtering, transforming and building are powerful transformations, but when they are _combined_ they become even more powerful!
+
+Let's play a game to explore what happens when we combine Transformers.
+
+[.lesson-instruction]
+--
+*Get Ready:*
+
+- Each pair of students needs one set of cut-out cards from @handout{create-transformer-cards-2.adoc}.
```

Comment:
I'd use the scaffolded version of the cards here, to remind them of what the inputs are.